notes/retrain_after_two_stages_pruning/parameters_and_log_for_effective_retrain_20240503_ver4DataSet.txt
====================================================================
###針對二次剪枝後的重訓練，這個設定滿有效果的，要實驗是哪個參數造成的效果。
epoch:
acc:
opt.batchSize = 64;
opt.LR = 0.1;
opt.weightDecay = 5e-4;
opt.momentum = 0.5;
opt.nEpochs = 500;
opt.schedule = [0.3, 0.6, 0.9];

===============================================================
save dir is: ../../../trained_models/step_5_retrain_after_step_4/train_after_second_pruning_prunratio0.85_2024050312/
***Use device:cuda:0
length of samples:651
config is [5, 32, 10, 8, 17, 18, 27, 39, 34, 41, 72, 3]
+----------------------------------------------------------------------------+
+                           Pytorch Model Summary                            +
------------------------------------------------------------------------------
   Layer (type)       Input Shape      Output Shape    Param #      FLOPS #
==============================================================================
       Conv2d-1     (1, 1, 30225)     (5, 1, 15109)         45      679,905
  BatchNorm2d-2     (5, 1, 15109)     (5, 1, 15109)         10            0
         ReLu-3     (5, 1, 15109)     (5, 1, 15109)          0       75,545
       Conv2d-4     (5, 1, 15109)     (32, 1, 7553)        800    6,042,400
  BatchNorm2d-5     (32, 1, 7553)     (32, 1, 7553)         64            0
         ReLu-6     (32, 1, 7553)     (32, 1, 7553)          0      241,696
    MaxPool2d-7     (32, 1, 7553)      (32, 1, 151)          0      241,600
      Permute-8      (32, 1, 151)      (1, 32, 151)          0            0
       Conv2d-9      (1, 32, 151)     (10, 32, 151)         90      434,880
 BatchNorm2d-10     (10, 32, 151)     (10, 32, 151)         20            0
        ReLu-11     (10, 32, 151)     (10, 32, 151)          0       48,320
   MaxPool2d-12     (10, 32, 151)      (10, 16, 75)          0       48,000
      Conv2d-13      (10, 16, 75)       (8, 16, 75)        720      864,000
 BatchNorm2d-14       (8, 16, 75)       (8, 16, 75)         16            0
        ReLu-15       (8, 16, 75)       (8, 16, 75)          0        9,600
      Conv2d-16       (8, 16, 75)      (17, 16, 75)      1,224    1,468,800
 BatchNorm2d-17      (17, 16, 75)      (17, 16, 75)         34            0
        ReLu-18      (17, 16, 75)      (17, 16, 75)          0       20,400
   MaxPool2d-19      (17, 16, 75)       (17, 8, 37)          0       20,128
      Conv2d-20       (17, 8, 37)       (18, 8, 37)      2,754      815,184
 BatchNorm2d-21       (18, 8, 37)       (18, 8, 37)         36            0
        ReLu-22       (18, 8, 37)       (18, 8, 37)          0        5,328
      Conv2d-23       (18, 8, 37)       (27, 8, 37)      4,374    1,294,704
 BatchNorm2d-24       (27, 8, 37)       (27, 8, 37)         54            0
        ReLu-25       (27, 8, 37)       (27, 8, 37)          0        7,992
   MaxPool2d-26       (27, 8, 37)       (27, 4, 18)          0        7,776
      Conv2d-27       (27, 4, 18)       (39, 4, 18)      9,477      682,344
 BatchNorm2d-28       (39, 4, 18)       (39, 4, 18)         78            0
        ReLu-29       (39, 4, 18)       (39, 4, 18)          0        2,808
      Conv2d-30       (39, 4, 18)       (34, 4, 18)     11,934      859,248
 BatchNorm2d-31       (34, 4, 18)       (34, 4, 18)         68            0
        ReLu-32       (34, 4, 18)       (34, 4, 18)          0        2,448
   MaxPool2d-33       (34, 4, 18)        (34, 2, 9)          0        2,448
      Conv2d-34        (34, 2, 9)        (41, 2, 9)     12,546      225,828
 BatchNorm2d-35        (41, 2, 9)        (41, 2, 9)         82            0
        ReLu-36        (41, 2, 9)        (41, 2, 9)          0          738
      Conv2d-37        (41, 2, 9)        (72, 2, 9)     26,568      478,224
 BatchNorm2d-38        (72, 2, 9)        (72, 2, 9)        144            0
        ReLu-39        (72, 2, 9)        (72, 2, 9)          0        1,296
   MaxPool2d-40        (72, 2, 9)        (72, 1, 4)          0        1,152
      Conv2d-41        (72, 1, 4)         (3, 1, 4)        216          864
 BatchNorm2d-42         (3, 1, 4)         (3, 1, 4)          6            0
        ReLu-43         (3, 1, 4)         (3, 1, 4)          0           12
   AvgPool2d-44         (3, 1, 4)         (3, 1, 1)          0           12
     Flatten-45         (3, 1, 1)            (1, 3)          0            0
      Linear-46            (1, 3)            (1, 3)         12           12
     Softmax-47            (1, 3)            (1, 3)          0            3
==============================================================================
Total Params: 71,372
Total FLOPs : 14,583,695
------------------------------------------------------------------------------
Input size (MB) : 0.12
Params size (MB): 0.27
Total size (MB) : 0.39
------------------------------------------------------------------------------

sfeb.0.weight:True
sfeb.1.weight:True
sfeb.1.bias:True
sfeb.3.weight:True
sfeb.4.weight:True
sfeb.4.bias:True
tfeb.0.weight:True
tfeb.1.weight:True
tfeb.1.bias:True
tfeb.4.weight:True
tfeb.5.weight:True
tfeb.5.bias:True
tfeb.7.weight:True
tfeb.8.weight:True
tfeb.8.bias:True
tfeb.11.weight:True
tfeb.12.weight:True
tfeb.12.bias:True
tfeb.14.weight:True
tfeb.15.weight:True
tfeb.15.bias:True
tfeb.18.weight:True
tfeb.19.weight:True
tfeb.19.bias:True
tfeb.21.weight:True
tfeb.22.weight:True
tfeb.22.bias:True
tfeb.25.weight:True
tfeb.26.weight:True
tfeb.26.bias:True
tfeb.28.weight:True
tfeb.29.weight:True
tfeb.29.bias:True
tfeb.33.weight:True
tfeb.34.weight:True
tfeb.34.bias:True
tfeb.38.weight:True
tfeb.38.bias:True
ACDNet model has been prepared for training
+----------------------------------------------------------------------------+
+                           Pytorch Model Summary                            +
------------------------------------------------------------------------------
   Layer (type)       Input Shape      Output Shape    Param #      FLOPS #
==============================================================================
       Conv2d-1     (1, 1, 30225)     (5, 1, 15109)         45      679,905
  BatchNorm2d-2     (5, 1, 15109)     (5, 1, 15109)         10            0
         ReLu-3     (5, 1, 15109)     (5, 1, 15109)          0       75,545
       Conv2d-4     (5, 1, 15109)     (32, 1, 7553)        800    6,042,400
  BatchNorm2d-5     (32, 1, 7553)     (32, 1, 7553)         64            0
         ReLu-6     (32, 1, 7553)     (32, 1, 7553)          0      241,696
    MaxPool2d-7     (32, 1, 7553)      (32, 1, 151)          0      241,600
      Permute-8      (32, 1, 151)      (1, 32, 151)          0            0
       Conv2d-9      (1, 32, 151)     (10, 32, 151)         90      434,880
 BatchNorm2d-10     (10, 32, 151)     (10, 32, 151)         20            0
        ReLu-11     (10, 32, 151)     (10, 32, 151)          0       48,320
   MaxPool2d-12     (10, 32, 151)      (10, 16, 75)          0       48,000
      Conv2d-13      (10, 16, 75)       (8, 16, 75)        720      864,000
 BatchNorm2d-14       (8, 16, 75)       (8, 16, 75)         16            0
        ReLu-15       (8, 16, 75)       (8, 16, 75)          0        9,600
      Conv2d-16       (8, 16, 75)      (17, 16, 75)      1,224    1,468,800
 BatchNorm2d-17      (17, 16, 75)      (17, 16, 75)         34            0
        ReLu-18      (17, 16, 75)      (17, 16, 75)          0       20,400
   MaxPool2d-19      (17, 16, 75)       (17, 8, 37)          0       20,128
      Conv2d-20       (17, 8, 37)       (18, 8, 37)      2,754      815,184
 BatchNorm2d-21       (18, 8, 37)       (18, 8, 37)         36            0
        ReLu-22       (18, 8, 37)       (18, 8, 37)          0        5,328
      Conv2d-23       (18, 8, 37)       (27, 8, 37)      4,374    1,294,704
 BatchNorm2d-24       (27, 8, 37)       (27, 8, 37)         54            0
        ReLu-25       (27, 8, 37)       (27, 8, 37)          0        7,992
   MaxPool2d-26       (27, 8, 37)       (27, 4, 18)          0        7,776
      Conv2d-27       (27, 4, 18)       (39, 4, 18)      9,477      682,344
 BatchNorm2d-28       (39, 4, 18)       (39, 4, 18)         78            0
        ReLu-29       (39, 4, 18)       (39, 4, 18)          0        2,808
      Conv2d-30       (39, 4, 18)       (34, 4, 18)     11,934      859,248
 BatchNorm2d-31       (34, 4, 18)       (34, 4, 18)         68            0
        ReLu-32       (34, 4, 18)       (34, 4, 18)          0        2,448
   MaxPool2d-33       (34, 4, 18)        (34, 2, 9)          0        2,448
      Conv2d-34        (34, 2, 9)        (41, 2, 9)     12,546      225,828
 BatchNorm2d-35        (41, 2, 9)        (41, 2, 9)         82            0
        ReLu-36        (41, 2, 9)        (41, 2, 9)          0          738
      Conv2d-37        (41, 2, 9)        (72, 2, 9)     26,568      478,224
 BatchNorm2d-38        (72, 2, 9)        (72, 2, 9)        144            0
        ReLu-39        (72, 2, 9)        (72, 2, 9)          0        1,296
   MaxPool2d-40        (72, 2, 9)        (72, 1, 4)          0        1,152
      Conv2d-41        (72, 1, 4)         (3, 1, 4)        216          864
 BatchNorm2d-42         (3, 1, 4)         (3, 1, 4)          6            0
        ReLu-43         (3, 1, 4)         (3, 1, 4)          0           12
   AvgPool2d-44         (3, 1, 4)         (3, 1, 1)          0           12
     Flatten-45         (3, 1, 1)            (1, 3)          0            0
      Linear-46            (1, 3)            (1, 3)         12           12
     Softmax-47            (1, 3)            (1, 3)          0            3
==============================================================================
Total Params: 71,372
Total FLOPs : 14,583,695
------------------------------------------------------------------------------
Input size (MB) : 0.12
Params size (MB): 0.27
Total size (MB) : 0.39
------------------------------------------------------------------------------

/home/ai/miniconda3/envs/acdnetenv/lib/python3.10/site-packages/tqdm/auto.py:21: TqdmWarning: IProgress not found. Please update jupyter and ipywidgets. See https://ipywidgets.readthedocs.io/en/stable/user_install.html
  from .autonotebook import tqdm as notebook_tqdm
device is :cuda:0
len of Y:472
/tmp/ipykernel_192889/3459676549.py:123: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  x = torch.tensor(x)
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
save model to ../../../trained_models/step_5_retrain_after_step_4/train_after_second_pruning_prunratio0.85_2024050312/retrained_model_after_second_pruning_ratio85.0_acc68.22034454345703_0th_epoch_20240503123940.pt
SP-[1] Epoch: 1/500 | Time: 0m07s (Train 0m06s  Val 0m01s) | Train: LR 0.010000000000000002  Loss 0.89  Acc 45.31% | Val: Loss nan  Acc(top1) 68.22% | HA 68.22@1
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 2/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.010000000000000002  Loss 0.60  Acc 37.36% | Val: Loss nan  Acc(top1) 67.37% | HA 68.22@1
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 3/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.010000000000000002  Loss 0.58  Acc 43.32% | Val: Loss nan  Acc(top1) 66.53% | HA 68.22@1
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 4/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.010000000000000002  Loss 0.59  Acc 39.49% | Val: Loss nan  Acc(top1) 68.22% | HA 68.22@1
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
save model to ../../../trained_models/step_5_retrain_after_step_4/train_after_second_pruning_prunratio0.85_2024050312/retrained_model_after_second_pruning_ratio85.0_acc69.06779479980469_4th_epoch_20240503124005.pt
SP-[1] Epoch: 5/500 | Time: 0m06s (Train 0m06s  Val 0m00s) | Train: LR 0.010000000000000002  Loss 0.58  Acc 43.75% | Val: Loss nan  Acc(top1) 69.07% | HA 69.07@5
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 6/500 | Time: 0m06s (Train 0m06s  Val 0m00s) | Train: LR 0.010000000000000002  Loss 0.58  Acc 44.32% | Val: Loss nan  Acc(top1) 65.25% | HA 69.07@5
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 7/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.010000000000000002  Loss 0.56  Acc 45.03% | Val: Loss nan  Acc(top1) 68.22% | HA 69.07@5
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
save model to ../../../trained_models/step_5_retrain_after_step_4/train_after_second_pruning_prunratio0.85_2024050312/retrained_model_after_second_pruning_ratio85.0_acc71.61016845703125_7th_epoch_20240503124023.pt
SP-[1] Epoch: 8/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.010000000000000002  Loss 0.57  Acc 46.45% | Val: Loss nan  Acc(top1) 71.61% | HA 71.61@8
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
save model to ../../../trained_models/step_5_retrain_after_step_4/train_after_second_pruning_prunratio0.85_2024050312/retrained_model_after_second_pruning_ratio85.0_acc72.88135528564453_8th_epoch_20240503124030.pt
SP-[1] Epoch: 9/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.010000000000000002  Loss 0.56  Acc 44.74% | Val: Loss nan  Acc(top1) 72.88% | HA 72.88@9
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 10/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.010000000000000002  Loss 0.55  Acc 49.15% | Val: Loss nan  Acc(top1) 72.03% | HA 72.88@9
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 11/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.58  Acc 41.05% | Val: Loss nan  Acc(top1) 69.49% | HA 72.88@9
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 12/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.59  Acc 44.18% | Val: Loss nan  Acc(top1) 64.83% | HA 72.88@9
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 13/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.58  Acc 41.76% | Val: Loss nan  Acc(top1) 67.37% | HA 72.88@9
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 14/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.57  Acc 48.86% | Val: Loss nan  Acc(top1) 70.76% | HA 72.88@9
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
save model to ../../../trained_models/step_5_retrain_after_step_4/train_after_second_pruning_prunratio0.85_2024050312/retrained_model_after_second_pruning_ratio85.0_acc74.5762710571289_14th_epoch_20240503124107.pt
SP-[1] Epoch: 15/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.52  Acc 52.13% | Val: Loss nan  Acc(top1) 74.58% | HA 74.58@15
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 16/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.51  Acc 50.43% | Val: Loss nan  Acc(top1) 31.78% | HA 74.58@15
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 17/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.52  Acc 51.28% | Val: Loss nan  Acc(top1) 72.88% | HA 74.58@15
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
save model to ../../../trained_models/step_5_retrain_after_step_4/train_after_second_pruning_prunratio0.85_2024050312/retrained_model_after_second_pruning_ratio85.0_acc75.4237289428711_17th_epoch_20240503124126.pt
SP-[1] Epoch: 18/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.52  Acc 51.85% | Val: Loss nan  Acc(top1) 75.42% | HA 75.42@18
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 19/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.50  Acc 52.56% | Val: Loss nan  Acc(top1) 41.53% | HA 75.42@18
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
save model to ../../../trained_models/step_5_retrain_after_step_4/train_after_second_pruning_prunratio0.85_2024050312/retrained_model_after_second_pruning_ratio85.0_acc76.27118682861328_19th_epoch_20240503124138.pt
SP-[1] Epoch: 20/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.50  Acc 49.29% | Val: Loss nan  Acc(top1) 76.27% | HA 76.27@20
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 21/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.50  Acc 54.69% | Val: Loss nan  Acc(top1) 74.58% | HA 76.27@20
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 22/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.48  Acc 55.82% | Val: Loss nan  Acc(top1) 75.00% | HA 76.27@20
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 23/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.50  Acc 53.12% | Val: Loss nan  Acc(top1) 39.41% | HA 76.27@20
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 24/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.49  Acc 52.13% | Val: Loss nan  Acc(top1) 72.03% | HA 76.27@20
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 25/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.49  Acc 51.14% | Val: Loss nan  Acc(top1) 76.27% | HA 76.27@20
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 26/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.47  Acc 54.97% | Val: Loss nan  Acc(top1) 75.85% | HA 76.27@20
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 27/500 | Time: 0m06s (Train 0m06s  Val 0m00s) | Train: LR 0.1  Loss 0.50  Acc 53.55% | Val: Loss nan  Acc(top1) 76.27% | HA 76.27@20
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 28/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.52  Acc 52.70% | Val: Loss nan  Acc(top1) 72.88% | HA 76.27@20
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 29/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.48  Acc 53.41% | Val: Loss nan  Acc(top1) 65.68% | HA 76.27@20
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 30/500 | Time: 0m06s (Train 0m06s  Val 0m00s) | Train: LR 0.1  Loss 0.49  Acc 53.69% | Val: Loss nan  Acc(top1) 36.86% | HA 76.27@20
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 31/500 | Time: 0m06s (Train 0m06s  Val 0m00s) | Train: LR 0.1  Loss 0.48  Acc 51.85% | Val: Loss nan  Acc(top1) 42.37% | HA 76.27@20
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 32/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.50  Acc 54.12% | Val: Loss nan  Acc(top1) 38.14% | HA 76.27@20
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 33/500 | Time: 0m06s (Train 0m06s  Val 0m00s) | Train: LR 0.1  Loss 0.49  Acc 51.42% | Val: Loss nan  Acc(top1) 73.73% | HA 76.27@20
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
save model to ../../../trained_models/step_5_retrain_after_step_4/train_after_second_pruning_prunratio0.85_2024050312/retrained_model_after_second_pruning_ratio85.0_acc76.69491577148438_33th_epoch_20240503124306.pt
SP-[1] Epoch: 34/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.48  Acc 56.96% | Val: Loss nan  Acc(top1) 76.69% | HA 76.69@34
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 35/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.49  Acc 52.98% | Val: Loss nan  Acc(top1) 73.73% | HA 76.69@34
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 36/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.47  Acc 54.40% | Val: Loss nan  Acc(top1) 75.00% | HA 76.69@34
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 37/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.48  Acc 54.83% | Val: Loss nan  Acc(top1) 76.27% | HA 76.69@34
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 38/500 | Time: 0m06s (Train 0m06s  Val 0m00s) | Train: LR 0.1  Loss 0.48  Acc 54.55% | Val: Loss nan  Acc(top1) 38.98% | HA 76.69@34
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 39/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.49  Acc 54.40% | Val: Loss nan  Acc(top1) 73.31% | HA 76.69@34
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
save model to ../../../trained_models/step_5_retrain_after_step_4/train_after_second_pruning_prunratio0.85_2024050312/retrained_model_after_second_pruning_ratio85.0_acc79.23728942871094_39th_epoch_20240503124344.pt
SP-[1] Epoch: 40/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.48  Acc 54.55% | Val: Loss nan  Acc(top1) 79.24% | HA 79.24@40
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 41/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.47  Acc 56.11% | Val: Loss nan  Acc(top1) 73.73% | HA 79.24@40
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 42/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.46  Acc 57.39% | Val: Loss nan  Acc(top1) 37.71% | HA 79.24@40
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 43/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.49  Acc 54.83% | Val: Loss nan  Acc(top1) 57.63% | HA 79.24@40
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 44/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.51  Acc 53.69% | Val: Loss nan  Acc(top1) 75.00% | HA 79.24@40
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 45/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.50  Acc 52.84% | Val: Loss nan  Acc(top1) 72.46% | HA 79.24@40
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 46/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.47  Acc 57.95% | Val: Loss nan  Acc(top1) 72.88% | HA 79.24@40
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 47/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.49  Acc 53.55% | Val: Loss nan  Acc(top1) 75.85% | HA 79.24@40
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236
SP-[1] Epoch: 48/500 | Time: 0m06s (Train 0m05s  Val 0m00s) | Train: LR 0.1  Loss 0.48  Acc 53.55% | Val: Loss nan  Acc(top1) 72.88% | HA 79.24@40
shape of y_pred:torch.Size([472, 3])
shape of y_target:torch.Size([472, 3])
after: len of y_pred:236, len of y_target:236

src/common/opts.py
import os
import argparse

def parse():
    parser = argparse.ArgumentParser(description='ACDNet Sound Classification');

    # General settings
    parser.add_argument('--netType', default='TLACDNet',  required=False);
    parser.add_argument('--data', default='../datasets/processed/',  required=False);
    parser.add_argument('--dataset', required=False, default='fold1_dataset.npz', choices=['10']);
    parser.add_argument('--BC', default=True, action='store_true', help='BC learning');
    parser.add_argument('--strongAugment', default=True,  action='store_true', help='Add scale and gain augmentation');
    #在ipynb中，不能使用parser.parse，要改用parser.parse_known_args()
    opt, unknown = parser.parse_known_args()
    #Leqarning settings
    opt.batchSize = 5;#64;
    opt.weightDecay = 5e-4;
    opt.momentum = 0.09;
    opt.nEpochs = 10;#2000;
    opt.LR = 0.01#0.1;
    opt.schedule = [0.03, 0.06, 0.09]#[0.3, 0.6, 0.9];
    opt.warmup = 10;

    #Basic Net Settings
    opt.nClasses = 6#50;
    opt.nFolds = 1;#5;
    opt.splits = 1#[i for i in range(1, opt.nFolds + 1)];
    opt.sr = 16000#20000;
    opt.inputLength = 30225;
    #Test data
    opt.nCrops = 5;

    return opt


def display_info(opt):
    print('+------------------------------+');
    print('| {} Sound classification'.format(opt.netType));
    print('+------------------------------+');
    print('| dataset  : {}'.format(opt.dataset));
    print('| nEpochs  : {}'.format(opt.nEpochs));
    print('| LRInit   : {}'.format(opt.LR));
    print('| schedule : {}'.format(opt.schedule));
    print('| warmup   : {}'.format(opt.warmup));
    print('| batchSize: {}'.format(opt.batchSize));
    print('| Splits: {}'.format(opt.splits));
    print('+------------------------------+');
